grpc/src/main/java/io/stargate/grpc/service/ExceptionHandler.java

```diff
@@ -72,11 +74,34 @@ public void handleException(Throwable throwable) {
     }
   }
 
+  /**
+   * @param status
+   * @param throwable
+   * @return generic error message for gRPC exceptions using gRPC Status and Exception thrown If the
+   *     status is null and throwable is a {@link StatusException} or {@link StatusRuntimeException}
+   *     it extracts its status. If the status is null and throwable is another instance type, it
+   *     will have the status code {@code Status.UNKNOWN}.
+   */
+  protected String getErrorMessage(@Nullable Status status, @Nonnull Throwable throwable) {
+    if (status == null) {
+      if (throwable instanceof StatusException) {
+        status = ((StatusException) throwable).getStatus();
+      } else if (throwable instanceof StatusRuntimeException) {
+        status = ((StatusRuntimeException) throwable).getStatus();
+      } else {
+        status = Status.UNKNOWN;
+      }
+    }
+
+    return status.withDescription(throwable.getMessage()).withCause(throwable).toString();
```

Comment:
what kind of a message this gives, can you post an example? Maybe this is too much, we could have simpler messages not depending on the status..

Reply:
hi @ivansenic the error log looks as below 

```
ERROR [mainIOThread-0] 2022-11-07 16:53:07,799 ExceptionHandler.java:273 - Status{code=ALREADY_EXISTS, description=Cannot add existing keyspace "test", cause=org.apache.cassandra.stargate.exceptions.AlreadyExistsException: Cannot add existing keyspace "test"
	at io.stargate.db.dse.impl.Conversion.toExternal(Conversion.java:450)
	at io.stargate.db.dse.impl.Conversion.convertInternalException(Conversion.java:589)
	at io.stargate.db.dse.impl.DsePersistence$DseConnection.convertExceptionWithWarnings(DsePersistence.java:611)
	at io.stargate.db.dse.impl.DsePersistence$DseConnection.lambda$executeRequest$3(DsePersistence.java:591)
	at io.reactivex.internal.observers.ConsumerSingleObserver.onError(ConsumerSingleObserver.java:46)
	at io.reactivex.internal.operators.single.SingleDoFinally$DoFinallyObserver.onError(SingleDoFinally.java:79)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
	at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback$FlatMapSingleObserver.onError(SingleFlatMap.java:116)
	at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback.onError(SingleFlatMap.java:90)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
	at io.reactivex.internal.observers.ResumeSingleObserver.onError(ResumeSingleObserver.java:51)
	at io.reactivex.internal.observers.ResumeSingleObserver.onError(ResumeSingleObserver.java:51)
	at io.reactivex.internal.disposables.EmptyDisposable.error(EmptyDisposable.java:78)
	at io.reactivex.internal.operators.single.SingleError.subscribeActual(SingleError.java:42)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at io.reactivex.internal.operators.single.SingleResumeNext$ResumeMainSingleObserver.onError(SingleResumeNext.java:80)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:48)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at io.reactivex.internal.operators.single.SingleResumeNext.subscribeActual(SingleResumeNext.java:39)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at io.reactivex.internal.operators.single.SingleDelayWithCompletable$OtherObserver.onComplete(SingleDelayWithCompletable.java:69)
	at io.reactivex.internal.disposables.EmptyDisposable.complete(EmptyDisposable.java:68)
	at io.reactivex.internal.operators.completable.CompletableEmpty.subscribeActual(CompletableEmpty.java:27)
	at io.reactivex.Completable.subscribe(Completable.java:2309)
	at io.reactivex.internal.operators.single.SingleDelayWithCompletable.subscribeActual(SingleDelayWithCompletable.java:36)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at io.reactivex.internal.operators.single.SingleDefer.subscribeActual(SingleDefer.java:43)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at org.apache.cassandra.utils.flow.RxThreads$1SubscribeOn.lambda$subscribeActual$0(RxThreads.java:51)
	at org.apache.cassandra.concurrent.TPCRunnable.run(TPCRunnable.java:101)
	at org.apache.cassandra.concurrent.IOScheduler$PooledTaskWorker.run(IOScheduler.java:319)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
	at org.apache.cassandra.utils.concurrent.InlinedThreadLocalThread.run(InlinedThreadLocalThread.java:251)
	at org.apache.cassandra.concurrent.IOThread.run(IOThread.java:46)
	Suppressed: org.apache.cassandra.exceptions.AlreadyExistsException: Cannot add existing keyspace "test"
		at org.apache.cassandra.cql3.statements.schema.CreateKeyspaceStatement.apply(CreateKeyspaceStatement.java:71)
		at org.apache.cassandra.schema.SchemaManager.apply(SchemaManager.java:818)
		at org.apache.cassandra.schema.MigrationManager.lambda$applyAndAnnounce$8(MigrationManager.java:385)
		at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
		... 26 more
```

- we could have the log as `ERROR [mainIOThread-0] 2022-11-07 16:53:07,799 ExceptionHandler.java:273 - Status{code=ALREADY_EXISTS, description=Cannot add existing keyspace "test"}` without the `cause` section?
- I had put in the status codes for easier correlation to client response and server exception

Reply:
Hmmmmh. I don't think we want stack traces for this specific example, as it's client error? We'd only want things that fail due to something server does wrong or doesn't handle. But not when caller tries to do something they cannot.